java-discoveryengine/google-cloud-discoveryengine/src/test/java/com/google/cloud/discoveryengine/v1alpha/ChunkServiceClientHttpJsonTest.java

```diff
@@ -0,0 +1,302 @@
+        ApiExceptionFactory.createException(
+            new Exception(), FakeStatusCode.of(StatusCode.Code.INVALID_ARGUMENT), false);
+    mockService.addException(exception);
+
+    try {
+      String name =
+          "projects/project-6658/locations/location-6658/dataStores/dataStore-6658/branches/branche-6658/documents/document-6658/chunks/chunk-6658";
+      client.getChunk(name);
+      Assert.fail("No exception raised");
+    } catch (InvalidArgumentException e) {
+      // Expected exception.
+    }
+  }
+
+  @Test
+  public void listChunksTest() throws Exception {
+    Chunk responsesElement = Chunk.newBuilder().build();
+    ListChunksResponse expectedResponse =
+        ListChunksResponse.newBuilder()
+            .setNextPageToken("")
+            .addAllChunks(Arrays.asList(responsesElement))
+            .build();
+    mockService.addResponse(expectedResponse);
+
+    DocumentName parent =
+        DocumentName.ofProjectLocationDataStoreBranchDocumentName(
+            "[PROJECT]", "[LOCATION]", "[DATA_STORE]", "[BRANCH]", "[DOCUMENT]");
+
+    ListChunksPagedResponse pagedListResponse = client.listChunks(parent);
+
+    List<Chunk> resources = Lists.newArrayList(pagedListResponse.iterateAll());
+
+    Assert.assertEquals(1, resources.size());
+    Assert.assertEquals(expectedResponse.getChunksList().get(0), resources.get(0));
+
+    List<String> actualRequests = mockService.getRequestPaths();
+    Assert.assertEquals(1, actualRequests.size());
+
+    String apiClientHeaderKey =
+        mockService
+            .getRequestHeaders()
+            .get(ApiClientHeaderProvider.getDefaultApiClientHeaderKey())
+            .iterator()
+            .next();
+    Assert.assertTrue(
+        GaxHttpJsonProperties.getDefaultApiClientHeaderPattern()
+            .matcher(apiClientHeaderKey)
+            .matches());
+  }
+
+  @Test
+  public void listChunksExceptionTest() throws Exception {
+    ApiException exception =
+        ApiExceptionFactory.createException(
+            new Exception(), FakeStatusCode.of(StatusCode.Code.INVALID_ARGUMENT), false);
+    mockService.addException(exception);
+
+    try {
+      DocumentName parent =
+          DocumentName.ofProjectLocationDataStoreBranchDocumentName(
+              "[PROJECT]", "[LOCATION]", "[DATA_STORE]", "[BRANCH]", "[DOCUMENT]");
+      client.listChunks(parent);
+      Assert.fail("No exception raised");
+    } catch (InvalidArgumentException e) {
+      // Expected exception.
+    }
+  }
+
+  @Test
+  public void listChunksTest2() throws Exception {
+    Chunk responsesElement = Chunk.newBuilder().build();
+    ListChunksResponse expectedResponse =
+        ListChunksResponse.newBuilder()
+            .setNextPageToken("")
+            .addAllChunks(Arrays.asList(responsesElement))
+            .build();
+    mockService.addResponse(expectedResponse);
+
+    String parent =
+        "projects/project-3187/locations/location-3187/dataStores/dataStore-3187/branches/branche-3187";
```

Comment:
There may be an issue with the parent path where part of the segment is missing as the rpc `ListChunks` has `"/v1alpha/{parent=projects/*/locations/*/dataStores/*/branches/*}/documents/*/chunks"` instead of `"/v1alpha/{parent=projects/*/locations/*/dataStores/*/branches/*/documents/*}/chunks"` 

https://github.com/googleapis/googleapis/blob/569bf5b0541481388a3763157c5447e65839743f/google/cloud/discoveryengine/v1alpha/chunk_service.proto#L54-L63

This is also causing an issue for python client library generation. See https://github.com/googleapis/google-cloud-python/pull/12433#discussion_r1518567513

Googlers see b/328843326 and b/328869136